layers/+spacemacs/spacemacs-layouts/funcs.el

```diff
@@ -315,6 +315,13 @@ perspectives does."
                    (mapcar 'persp-kill
                            (helm-marked-candidates))))))))
 
+
+(defun spacemacs//helm-persp-switch-project-helper (project func)
+  (let ((persp-reset-windows-on-nil-window-conf t))
+    (persp-switch project)
```

Comment:
Would it make sense to create a new workspace in the layout, if the layout already exists? I mean something like the following:
```diff
                            (helm-marked-candidates))))))))

 (defun spacemacs//helm-persp-switch-project-helper (project func)
-  (let ((persp-reset-windows-on-nil-window-conf t))
+  (let ((persp-reset-windows-on-nil-window-conf t)
+        (persp-already-exists (persp-p (persp-get-by-name project))))
     (persp-switch project)
+    (when persp-already-exists
+      (eyebrowse-create-window-config))
     (funcall func project)
     (delete-other-windows)))
```
This would avoid clobbering the existing workspace.  On the other hand, `SPC w u` should also undo said clobbering, and maybe that is good enough.